athena-elasticsearch/src/main/java/com/amazonaws/athena/connectors/elasticsearch/ElasticsearchMetadataHandler.java

```diff
@@ -296,6 +294,29 @@ public GetSplitsResponse doGetSplits(BlockAllocator allocator, GetSplitsRequest
         return new GetSplitsResponse(request.getCatalogName(), splits);
     }
 
+    /**
+     * Get DataStream names from ES/OS cluster if supported.
+     * return empty if not supported.
+     * Notes: AWS's version of ES doesn't support data stream, it is a feature only available in the Elasticsearch distributed by Elastic itself, licensed under the Elastic License.
+     * @param client
+     * @return
+     */
+    private Stream<String> getDataStreamNamesFromClient(AwsRestHighLevelClient client)
+    {
+        try {
+            return client.indices().getDataStream(new GetDataStreamRequest("*"), RequestOptions.DEFAULT).getDataStreams()
+                    .stream()
+                    .map(dataStream -> dataStream.getName());
+        }
+        // gracefully exit for AWS ES.
+        // AWS's version of ES doesn't support data stream, it is a feature only available in the Elasticsearch distributed by Elastic itself, licensed under the Elastic License.
+        // https://stackoverflow.com/questions/68006918/aws-elasticsearch-service-and-datastream
```

Comment:
you don't need these comments because you already explained this in your method header.